pkg/api/graph/deployment.go

```diff
@@ -0,0 +1,415 @@
+package graph
+
+import (
+	"sort"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/search"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	//build "github.com/openshift/origin/pkg/build/api"
+	deploy "github.com/openshift/origin/pkg/deploy/api"
+	image "github.com/openshift/origin/pkg/image/api"
+)
+
+// DeploymentFlowMap describes a single deployment config and the objects
+// that contributed to that deployment.
+type DeploymentFlowMap map[*DeploymentConfigNode][]ImageDeploymentFlow
```

Comment:
we've talked about calling these pipelines in the UI, easier to find the code in the API if the terminology is consistent :)